### ISSUE TITLE
[Feature Request]: Alternate driver skins (rivals?)

### ISSUE THREAD

**Issue by Tenome** [emoose/OutRun2006Tweaks#78]
### Describe the feature you'd like to see added

According to Howard's notes from OutrunX, obj_chr_aut contains alternate costumes for the driver that might be from rivals or from multiplayer. Would it be possible to add these as a selection/random choice?

On a related note, FXT also let you drive as Clarissa and Flag Man, but they were janky from what I remember.

**Comment by lance8765** [comment]
Sam and Wolf were in C2C? This is new to me.

(i didn't even know the main guy was named Alberto btw XD)

**Comment by espiox** [comment]
At one point in development they planned to let you choose your driver (from the three guys and three girls) and I believe there’s unfinished files for a character select on the license screen?

**Comment by emoose** [comment]
The addr I mentioned in free cam thread can let you switch between a couple chars, think it gets written into savegame as well: `7C2400 (4 bytes)`

IIRC there were some pretty weird ones there as well, maybe some sort of test characters?
Wasn't able to get the girl chars to work right though for some reason, don't they show up as rivals? guess maybe they aren't coded to work in player car properly.

Wonder how far they got with character selection, maybe there's some leftovers for it in the code...

**Comment by emoose** [comment]
Seem to have the character choice unlocked properly now, instead of forcing it to the screen like the above:

![image](https://github.com/user-attachments/assets/9a7a3e1e-ec88-4982-a067-043bc457e232)

If anyone wants to try here's a build: 
[OutRun2006Tweaks-CharaSelectionUnlocker.zip](https://github.com/user-attachments/files/16460966/OutRun2006Tweaks-CharaSelectionUnlocker.zip)

Maybe there's a chance the girls can work on certain stages/modes, otherwise not really sure if this would be worth adding to Tweaks...

**Comment by envido32** [comment]
> If anyone wants to try here's a build: [OutRun2006Tweaks-CharaSelectionUnlocker.zip](https://github.com/user-attachments/files/16460966/OutRun2006Tweaks-CharaSelectionUnlocker.zip)
> 
> Maybe there's a chance the girls can work on certain stages/modes, otherwise not really sure if this would be worth adding to Tweaks...

I haven't been able to load this built. I tried with a clean installation and with admin privileges. v0.5a and v0.4.2 works just fine....

![image](https://github.com/user-attachments/assets/ebc57555-2ab7-414b-b2a2-7f6d63d49daa)

But I did spend some time trying the models by editing the SaveFile: Wolf seems to work just fine, Sam has a small animation issue on his legs... not a big deal but noticeable.

The girls as drivers on the other hand has their animations all broken. It doesn't crash but it looks awful... but they seem to have the same kind of "error", so my speculation is that fix one would fix all of the girls... if possible.

I will keep looking to see if the animations can be swaped or something... maybe some extra testing on different game modes and how it affects online for example.

**Comment by emoose** [comment]
Hm it still seems to run fine for me, that error would usually mean an issue with setting up the patches/hooks themselves, do you get any outrun2006tweaks.log file from it? Also are you using the OR2006C2C.exe from the tweaks DL?

**Comment by emoose** [comment]
Ah there were a couple extra changes in that build which could have broken something, here's a new build with just the license screen stuff:
[OutRun2006Tweaks-CharaSelectionUnlockerv2.zip](https://github.com/user-attachments/files/16563733/OutRun2006Tweaks-CharaSelectionUnlockerv2.zip)

**Comment by lance8765** [comment]
I've always wondered who's driving the rival cars in OutRun Mode. Do we actually see the models?

**Comment by envido32** [comment]
> Ah there were a couple extra changes in that build which could have broken something, here's a new build with just the license screen stuff: [OutRun2006Tweaks-CharaSelectionUnlockerv2.zip](https://github.com/user-attachments/files/16563733/OutRun2006Tweaks-CharaSelectionUnlockerv2.zip)

Weird... same error.
I uploaded the log file if you want to take a look

[OutRun2006Tweaks.log](https://github.com/user-attachments/files/16570958/OutRun2006Tweaks.log)

**Comment by emoose** [comment]
Hm, seems it's not even getting to apply the hooks, weird.

Just noticed there was some debug logging stuff in that code too, maybe that can cause issues on some systems.. 

Maybe see if this is any better: 
[OutRun2006Tweaks-CharaSelectionUnlockerv3.zip](https://github.com/user-attachments/files/16571063/OutRun2006Tweaks-CharaSelectionUnlockerv3.zip)

**Comment by Tenome** [comment]
I can't test it because it has that other issue that you already fixed, but it looks like it still works. Can you implement it in the main branch? I brought the driver selection screen but then didn't notice that it wasn't actually made an official feature yet lol.

![image](https://github.com/user-attachments/assets/243fc2f5-6a3d-4c00-bce9-4899a7be3526)

**Comment by lance8765** [comment]
I wonder if it'll be possible to change the colors of default in-game models, since TCRF mentions leftovers of a redheaded Jennifer and a brunette Clarissa.

**Comment by Tenome** [comment]
> I wonder if it'll be possible to change the colors of default in-game models, since TCRF mentions leftovers of a redheaded Jennifer and a brunette Clarissa.

Yes, since that's just a texture replacement. The models aren't ripped though so you'll have to do that yourself (NinjaRipper might work).

**Comment by Tenome** [comment]
> I wonder if it'll be possible to change the colors of default in-game models, since TCRF mentions leftovers of a redheaded Jennifer and a brunette Clarissa.
Swap the filenames of the textures and boom, redhead.

![OR2006C2C_n3WGdvfnsV](https://github.com/user-attachments/assets/a83414ed-d5bd-4165-b968-725779aa2868)

**Comment by lance8765** [comment]
> > I wonder if it'll be possible to change the colors of default in-game models, since TCRF mentions leftovers of a redheaded Jennifer and a brunette Clarissa.
> 
> Swap the filenames of the textures and boom, redhead.
> 
> ![OR2006C2C_n3WGdvfnsV](https://private-user-images.githubusercontent.com/12216081/364303466-a83414ed-d5bd-4165-b968-725779aa2868.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU0NDc3MzksIm5iZiI6MTcyNTQ0NzQzOSwicGF0aCI6Ii8xMjIxNjA4MS8zNjQzMDM0NjYtYTgzNDE0ZWQtZDViZC00MTY1LWI5NjgtNzI1Nzc5YWEyODY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA0VDEwNTcxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5MTgyNjg1OTQyMDZiMTk3YzU3NDBkN2RiZmY5NTc4YTFjZDY0OTBiNGM5NzZlNGQzMzM0YTZlNTUyMDhiODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4wjybHtr-QrocTGLw4r1BobgeywUQVKjV-Hjw_-Bhdc)

The WHAT, so it wasn't really a "leftover"!

Is brunette Clarissa also available?

**Comment by Tenome** [comment]
Do you mean brunette Jennifer? Yep.
![OR2006C2C_FDyuSTQeGV](https://github.com/user-attachments/assets/ca8898dc-5d13-4c48-9cdf-2dad74c02dec)

**Comment by envido32** [comment]
Wow! This is really interesting!

But if the files are not dumped, where did you ripped from? Are they insisde the game like JP's Clarrissa? In that case... could it be possible to have a toggle in the .ini file to select "Alternative GF"?

**Comment by Tenome** [comment]
The files were dumped via Tweaks, so they're in the game. They're just not used. They're in the same folder as the regular textures.

**Comment by Tenome** [comment]
The files for all driver/passenger skins are in obj_driver_rival_pmt. Swapping them out is simple, but it'd be cool to have it as a Tweaks option too.
